packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,670 @@
+      },
+    ],
+    [
+      "Variant Weight",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.weight?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Length",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.length?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Width",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.width?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Height",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.height?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant HS Code",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.hs_code?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Origin Country",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.origin_country?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Mid Code",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.mid_code?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+    [
+      "Variant Material",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.material?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+  ])
+
+  private readonly NEWLINE_ = "\r\n"
+  private readonly DELIMITER_ = ";"
+  private readonly BATCH_SIZE = 50
+
+  constructor({
+    manager,
+    batchJobService,
+    productService,
+    fileService,
+    productRepository,
+  }: InjectedDependencies) {
+    super({
+      manager,
+      batchJobService,
+      productService,
+      fileService,
+      productRepository,
+    })
+
+    this.manager_ = manager
+    this.batchJobService_ = batchJobService
+    this.productService_ = productService
+    this.fileService_ = fileService
+    this.productRepository_ = productRepository
+  }
+
+  async buildTemplate(): Promise<string> {
+    return this.buildHeader()
+  }
+
+  async prepareBatchJobForProcessing(
+    batchJob: AdminPostBatchesReq,
+    req: Request
+  ): Promise<AdminPostBatchesReq> {
+    const { limit, offset, order, fields, expand, ...context } =
+      batchJob.context
+
+    const {
+      select: filterableFields,
+      skip,
+      take,
+      ...listConfig
+    } = prepareListQuery(
+      {
+        context: batchJob.context,
+        type: batchJob.type,
+        dry_run: batchJob.dry_run,
+        limit: limit as number,
+        offset: offset as number,
+        order: order as string,
+        fields: fields as string,
+        expand: expand as string,
+      },
+      {
+        isList: true,
+        defaultRelations: this.relations_,
```

Comment:
comment: To be modified if other suggestion is incorporated
```suggestion
        defaultRelations: this.defaultRelations_,
```